### DISCUSSION TITLE
[ROCm] Running SD.Next on RDNA 1 (RX 5000 series) cards

### DISCUSSION THREAD

**Discussion by cl0ck-byte** [vladmandic/sdnext#2005] [General]
This thread is for keeping up info about ROCm/PyTorch support of these cards.
**Please, do not bother SD.Next devs about RDNA1 support - bother AMD/PyTorch instead.** (see https://github.com/pytorch/pytorch/issues/106728)

Don't know how to install ROCm? [AMD's official guide](https://rocm.docs.amd.com/en/latest/deploy/linux/prerequisites.html) should cover you. Remember to use single-version installation of HIP SDK and don't forget about adding yourself to video and render groups after installing ROCm.
#
Since SD.Next is running `torch>=2.0.0`, RDNA1 cards will not work by default - manual installation is required.

For now, downgrading torch/torchivsion to versions pre 2.0.0 with
`pip install torch==1.13.1+rocm5.2 torchvision==0.14.1+rocm5.2 --index-url https://download.pytorch.org/whl/rocm5.2`

and forcing enviroment variable
`export HSA_OVERRIDE_GFX_VERSION=10.3.0`

and running SD.Next with flag
`--experimental`

seems to work, although with limited functionality (SDXL is probably not supported).
**Also cards have to run in full precision and with `--no-half` in order to generate anything** - both of these settings can be configured in webui.

Upcast sampling doesn't seem to work - generates monochrome solid colors.

**Comment by cl0ck-byte** [comment]
Speaking from the future: someone is cooking RDNA1 support finally for ROCm 6.2 but it's not AMD, so things might finally change

ROCm/Tensile/pull/1916

**Comment by daniandtheweb** [comment]
Recently I've found about this project https://github.com/lamikr/rocm_sdk_builder; it allows to build the whole ROCm stack for the RDNA1 family including pytorch.
I've been using it for some weeks now and everything works fine in fp16 without any GFX override or any other workaround.
